azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_azurerm.go

```diff
@@ -147,35 +253,38 @@ func expandServiceEndpointAzureRM(d *schema.ResourceData) (*serviceendpoint.Serv
 		(*serviceEndpoint.Data)["managementGroupName"] = d.Get("azurerm_management_group_name").(string)
 	}
 
-	if _, ok := d.GetOk("credentials"); ok {
-		credentials := d.Get("credentials").([]interface{})[0].(map[string]interface{})
-		(*serviceEndpoint.Authorization.Parameters)["serviceprincipalid"] = credentials["serviceprincipalid"].(string)
-		(*serviceEndpoint.Authorization.Parameters)["serviceprincipalkey"] = credentials["serviceprincipalkey"].(string)
-		(*serviceEndpoint.Data)["creationMode"] = "Manual"
-	}
-
 	serviceEndpoint.Type = converter.String("azurerm")
 	serviceEndpoint.Url = converter.String(endpointUrl)
 	return serviceEndpoint, projectID, nil
 }
 
-func flattenCredentials(d *schema.ResourceData, serviceEndpoint *serviceendpoint.ServiceEndpoint, hashKey string, hashValue string) interface{} {
+func flattenCredentials(d *schema.ResourceData, serviceEndpoint *serviceendpoint.ServiceEndpoint, hashKey string, hashValue string, serviceEndPointType AzureRmEndpointType) interface{} {
 	// secret value won't return by service and should not be overwritten
-	return []map[string]interface{}{{
-		"serviceprincipalid":  (*serviceEndpoint.Authorization.Parameters)["serviceprincipalid"],
-		"serviceprincipalkey": d.Get("credentials.0.serviceprincipalkey").(string),
-		hashKey:               hashValue,
-	}}
+	if serviceEndPointType == WorkloadIdentityFederation {
+		return []map[string]interface{}{{
+			"serviceprincipalid":  (*serviceEndpoint.Authorization.Parameters)["serviceprincipalid"],
+			"serviceprincipalkey": d.Get("credentials.0.serviceprincipalkey").(string),
```

Comment:
There is no `serviceprincipalkey` for WIF. Why are we setting it? Then if someone has set it up, ADO would reject it.

Reply:
Although it is being set here, it is just for the purposes of state management. It is not actually sent to the API.